src/Microsoft.TemplateSearch.TemplateDiscovery/PackProviders/TestPackProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.TemplateSearch.TemplateDiscovery.PackProviders
+{
+    internal class TestPackProvider : IPackProvider
+    {
+        private readonly string _folder;
+
+        internal TestPackProvider(string folder)
+        {
+            if (string.IsNullOrWhiteSpace(folder))
+            {
+                throw new ArgumentException($"'{nameof(folder)}' cannot be null or whitespace.", nameof(folder));
+            }
+            _folder = folder;
+        }
+
+        public string Name => "TestProvider";
+
+        public void DeleteDownloadedPacks()
+        {
+            //do nothing - do not remove test packs
+        }
+
+        public Task<IDownloadedPackInfo?> DownloadPackageAsync(IPackInfo packinfo)
+        {
+            return Task.FromResult((IDownloadedPackInfo?)packinfo);
+        }
+
+        public async IAsyncEnumerable<IPackInfo> GetCandidatePacksAsync()
```

Comment:
nice technique in this method

Reply:
consider the use of cancellation tokens in async methods you introduce, it's a good way to ensure such tasks don't infinitely hang, and almost a must-have if methods you invoke accept them.

Reply:
that's a nice catch, I think it's a good time to make the app cancellable giving it takes long time to execute.